File: strings/pallindrome.js
function isPalindrome(str) {
    let st = 0;
    let end = str.length - 1;
    while (st < end) {
        if (str[st++] !== str[end--]) {
            alert("Not a valid palindrome");
            return false;
        }
    }
    alert("Valid palindrome");
    return true;
}

function main() {
    let word = prompt("Enter a word: ");
    isPalindrome(word);
}

main();

File: strings/pallindrome.cpp
#include<iostream>
#include<cstring>
using namespace std;
bool isPallindrome(char str[],int n)
{
    int st=0,end=n-1;
  while(st<end)
  {
  if(str[st++]!=str[end--])
  {
    cout<<"Not valid pallindrome";
    return false;
  }
  }
  cout<<"Valid pallindrome";
}
int main()
{
  char word[100];  // Allocate enough space for the input string
    cout << "Enter a word: ";
    cin >> word;
    isPallindrome(word, strlen(word));
    return 0;

}